src/rgw/rgw_sal_motr.cc

```diff
@@ -3744,22 +3737,24 @@ int MotrMultipartUpload::list_parts(const DoutPrefixProvider *dpp, CephContext *
     if(ret_rc < 0)
       return ret_rc;
 
-    key_name = ent.key.name + "[" + ent.key.instance + "]";
-
-    // fetch the version-id in case of null version-id
-    if(ent.key.instance == "null")
+    if(!ent.is_delete_marker())
```

Comment:
- Delete marker is considered as a simple object, which we add in bucket index and not as a multipart object. 
- Delete marker deletion case is directly handled in delete_obj() method, So, in that case, we are not coming to this list_parts() method.
- If user tries to delete any specific object version and if the delete marker is the latest entry of an object. Then in the list_parts() method, we always get the latest entry (i.e delete marker). Hence to avoid delete marker entry and get the specific version, this check is added.

Reply:
thanks for clarifying